app/Http/Controllers/TradeController.php

```diff
@@ -49,7 +50,15 @@ public function store(TradeStoreRequest $request)
      */
     public function show(Request $request, Trade $trade)
     {
-        return view('trade.show', compact('trade'));
+        $currentUserID = Auth::user()->id;
```

Comment:
Little nit-picky but you can just do 
```suggestion
        $currentUserID = Auth::id();
```
here :)